src/App/Repository/Form/StatsRepository.php

```diff
@@ -260,12 +281,62 @@ public function getRecipients($form_id)
         ];
         $query = $this->selectQuery($where)
             ->resetSelect()
-            ->select([DB::expr('COUNT(distinct contact_id)'), 'total']);
+            ->select([DB::expr('COUNT(contacts.id)'), 'total']);
+
         $query = $this->targetedSurveyStateJoin($query);
+        if ($created_after || $created_before) {
```

Comment:
I am thinking that we may need this $query to be outside the `  if ($created_after || $created_before) {`  since you still want to do the join with messages and filter only outgoing even without a created_after/before filter .

Reply:
I think that because (when we are not filtering by date) we just want the total number of recipients, we're just looking up contacts for that targeted Survey in the query in `targetedSurveyStateJoin`. But if we want to filter by date, we need to join the messages table so we can only get the contacts for messages sent in that date range.

Reply:
I think you're correct. My impression was that maybe that was a bug caused by me when I coded that originally but now that I reviewed the other joins I think its actually correct. Thanks !